bundles/org.openhab.binding.weathercompany/README.md

```diff
@@ -146,13 +146,14 @@ The following channels will be translated to local language based on the languag
 ### Thing Example
 
 ```
-Bridge weathercompany:account:bridge [ apiKey="734982347982374" ] {
-    Thing weathercompany:weather-forecast:forecast "Local Forecast" @ "Home" [locationType="postalCode",postalCode="10001:US",language="en-US",refreshInterval=30]
-    Thing weathercompany:weather-forecast:chitown "Chicago Forecast" @ "Ohare Airport" [apiKey="734982347982374",locationType="iataCode",iataCode="ORD",language="en-US",refreshInterval=30]
-    Thing weathercompany:weather-forecast:miami "Miami Weather" @ "South Beach" [locationType="postalCode",postalCode="33139:US",language="es-US",refreshInterval=30]
-    Thing weathercompany:weather-observations:observations "Local Observations" @ "Home" [pwsStationId="KFLMIAMI208",refreshInterval=30]
-    Thing weathercompany:weather-observations:patagonia "Torres del Paine Weather" @ "Patagonia" [pwsStationId="IPUNTAAR4",refreshInterval=30]
-}
+Bridge weathercompany:account:bridge [ apiKey="0123456789" ]
+
+Thing weathercompany:weather-forecast:forecast "Local Forecast" (weathercompany:account:bridge) @ "Home" [locationType="postalCode",postalCode="10001:US",language="en-US",refreshInterval=30]
```

Comment:
I do not understand the idea behind this. Yes, the syntax is valid but imo the following is easier to understand:

```
Bridge weathercompany:account:bridge [ apiKey="0123456789" ]
    Thing weather-forecast forecast "Local Forecast" @ "Home" [locationType="postalCode",postalCode="10001:US",language="en-US",refreshInterval=30]	
    ....
}
```

Reply:
Yes, that's true. However, the items example won't work with the things example because of how the Thing UID is formed when using the `Bridge { Thing }` approach. 

By doing it this way, the items example will work whether you add the things using Paper UI, as well as with a .things file. It's the only way I could figure out how to make it work both ways.

And, it is supported by the openHAB [documentation](https://www.openhab.org/docs/configuration/things.html#defining-bridges-using-files).

Reply:
I should add that it's not a dealbreaker for me if you want to keep it using the `Bridge { Thing }` approach. I was just trying to make it easy for users who use Paper UI and .things files.

Reply:
The fact I am worried about at most is that you want to use a `ThingUID` without the bridge ID as third segment.

Is that a nasty bug of Paper UI when adding discovered Things having a Bridge? Or is this a more general problem of the discovery / inbox features? Do you have a clue?

maybe is will be fixed if you add it in the bindings discovery service (see e.g. [DarkSky](https://github.com/openhab/openhab-addons/blob/666bfe53fa806c3c51ca1fbb11c6c71c9df582f5/bundles/org.openhab.binding.darksky/src/main/java/org/openhab/binding/darksky/internal/discovery/DarkSkyDiscoveryService.java#L132)):

https://github.com/openhab/openhab-addons/blob/666bfe53fa806c3c51ca1fbb11c6c71c9df582f5/bundles/org.openhab.binding.weathercompany/src/main/java/org/openhab/binding/weathercompany/internal/discovery/WeatherCompanyDiscoveryService.java#L120

Change it to:

```java
        ThingUID localWeatherThing = new ThingUID(THING_TYPE_WEATHER_FORECAST, bridgeHandler.getThing().getUID(), LOCAL);
```

Reply:
It's completely unrelated to discovery. It happens when I add a thing manually using Paper UI.

So, for example, if I add a WeatherCompany thing manually using Paper UI (in the UI I I gave it a Thing ID of `test` and selected an existing Bridge Thing), it creates a thing whose Thing UID is `weathercompany:weather-forecast:test`.

Also, I'm not sure I see any functional difference between the existing code, and your suggested change. In both cases, `withBridge` is being called with the BridgeUID.

Reply:
From my POV that's wrong. When creating a Thing related to a Bridge the resulting `ThingUID` should contain the Bridge Id.

ThingUID w/o Bridge:

```
<bindingId>.<thingTypeId>.<thingId>
```

`ThingUID` with Bridge:

```
<bindingId>.<thingTypeId>.<bridgeId>.<thingId>
```

My suggested change adds the bridge Id to the `ThingUID`:

```java
        ThingUID localWeatherThing = new ThingUID(THING_TYPE_WEATHER_FORECAST, bridgeUID, LOCAL);
```

Reply:
Ah, I see what's different in your suggested change.

I'm happy to make that change in the binding, but I don't see how that changes what Paper UI does when you manually add a thing. The binding discovery code never gets executed in that case, because Paper UI does all the thing creation using the REST API. Right?

Reply:
Unfortunately, yes. Adding such Things manually via Paper UI results in wrong `ThingUID`s. IMO we should not care much about Paper UI anymore. Hopefully the new UI does it better.

I'd like to have @kaikreuzer opinion on this. Should we force `ThingUID`s containing a Bridge Id or not?

Reply:
Makes sense. I'll do whatever you guys want. It's just that the current documentation is wrong, and I'm anxious to get it fixed one way or the other.

Reply:
I can understand @cweitkamp's confusion here, because my assumption was as well that the bridgeId(s) is/are always a mandatory part of the ThingUID.
The `ThingUID` class has the (not deprecated) method `getBridgeIds`, so this also hints at the fact that it is expected to contain those.
Technically, it is right that the Paper UI does not require to include the bridge id(s) in the ThingUID, but I agree that it should better do so.
So the question goes to @ghys here: Does the new UI create Things that have their bridgeID as part of the newly created Thing? If not, it would be lovely if it could be changed that way :-).

Reply:
Thanks for the explanation @kaikreuzer. 

If it's a requirement that the ThingUID contain the Bridge Id, then shouldn't the framework enforce that requirement. But then, what would happen if the Bridge thing is deleted? Wouldn't the remaining things ThingUIDs contain a bridge Id that no longer exists? Or, does the framework currently not allow you to delete a Bridge Thing if there are Things that reference it?

Irrespective of the above, I'll change my PR so that the ThingUIDs contain the Bridge Id.

Reply:
Things where the bridge is deleted are still defined, but will stay UNINITIALIZED. Once a bridge turns up, everything is fine.

Reply:
> So the question goes to @ghys here: Does the new UI create Things that have their bridgeID as part of the newly created Thing?

No it does not, I didn't know about that convention. The problem is, the user can delay assigning the bridge creating the thing, it's not mandatory to specify it at creation time, and it can even be updated later (and you can even switch a Thing from one bridge to another, not a common scenario, I'll grant you that, but I have actually done it once or twice to switch MQTT things from one broker to another). Note that Paper UI allowed that too IIRC.

Reply:
> Things where the bridge is deleted are still defined, but will stay UNINITIALIZED. Once a bridge turns up, everything is fine.

I understand that. And, maybe I'm missing something. But, after a new bridge is created, there's no guarantee that the thing's ThingUID will contain the bridge Id of the new bridge it now references. The thing's ThingUID will contain the Bridge Id of the bridge that was deleted, which may or may not be the Bridge Id of the bridge it now references.

Reply:
Yes, I think @ghys is saying what I was trying to say, albeit, I think he said it more clearly.

Reply:
That's a good argument, @ghys. The ThingUID should not be touched if the user reassigns it to another bridge. So we would have to stick to the same convention as we do for the thing type uid within the ThingUID: It is recommended to use it (especially when textually defining a thing), but there must not be any semantics being assumed within a ThingUID.

Reply:
To cut the discussion short: As we are talking about the textual configuration here, I very much support @cweitkamp's few that it should be done with the nested syntax, because this way openHAB will automatically make sure that the things reference the correct bridge and no typos can be in the way.

Reply:
FYI I'm now noticing that for instance, the Hue binding seems to assign UIDs containing the bridge UID and a device ID, both of which can be changed:

![image](https://user-images.githubusercontent.com/2004147/82050722-f06c4e00-96b8-11ea-83f7-c37b6f96ef68.png)

Reply:
Thank you all for wrapping this up and clarify on the topic.

> but there must not be any semantics being assumed within a ThingUID.

I think we can and should make one exception: automatic discovery. This is a framework feature and in this case the framework can and  should force the recommended semantic. Thus I propose to add a small check in the `DiscoveryResultBuilder#withBridge()` to validate on related BridgeUID and ThingUIDs.

Reply:
Yes, what I meant was: Naming convention to define ThingUIDs (e.g. textually or by discovery) exactly this way (using thing type and bridge ids as segments), while nowhere in the code to expect being able to read any semantics out of a ThingID (because the user might have changed the bridge or the binding migrated the thing type to a different one or whatever might have been happened).